common-service/src/main/java/org/sunbird/util/authentication/RequestInterceptor.java

```diff
@@ -0,0 +1,115 @@
+package org.sunbird.util.authentication;
+
+import java.util.List;
+import org.apache.commons.lang3.StringUtils;
+import org.sunbird.common.models.util.JsonKey;
+import org.sunbird.common.request.HeaderParam;
+import play.mvc.Http;
+import play.mvc.Http.Request;
+
+/**
+ * This class will do the request header validation
+ *
+ * @author Amit Kumar
+ */
+public class RequestInterceptor {
+
+  private static List<String> restrictedUrlList;
+  private static List<String> publicUrlList;
+  private static List<String> privateUrlList;
+
+  public static List<String> getRestrictedUrlList() {
+    return restrictedUrlList;
+  }
+
+  public static void setRestrictedUrlList(List<String> restrictedUrlList) {
+    RequestInterceptor.restrictedUrlList = restrictedUrlList;
+  }
+
+  public static List<String> getPublicUrlList() {
+    return publicUrlList;
+  }
+
+  public static void setPublicUrlList(List<String> publicUrlList) {
+    RequestInterceptor.publicUrlList = publicUrlList;
+  }
+
+  public static List<String> getPrivateUrlList() {
+    return privateUrlList;
+  }
+
+  public static void setPrivateUrlList(List<String> privateUrlList) {
+    RequestInterceptor.privateUrlList = privateUrlList;
+  }
+
+  private RequestInterceptor() {}
+
+  /**
+   * This Method will do the request header validation.
+   *
+   * @param ctx Request
+   * @return String
+   */
+  public static String verifyRequestData(Http.Context ctx) {
+    Request request = ctx.request();
+    if (!isRequestInExcludeList(request.path())) {
+      String clientId = JsonKey.UNAUTHORIZED;
+      String userToken = request.getHeader(HeaderParam.X_Authenticated_User_Token.getName());
+      String authClientToken =
+          request.getHeader(HeaderParam.X_Authenticated_Client_Token.getName());
+      String authClientId = request.getHeader(HeaderParam.X_Authenticated_Client_Id.getName());
+
+      if (StringUtils.isNotBlank(userToken)) {
+        clientId = AuthenticationHelper.verifyUserToken(userToken);
+      } else if (StringUtils.isNotBlank(authClientToken) && StringUtils.isNotBlank(authClientId)) {
+        clientId = AuthenticationHelper.verifyClientAccessToken(authClientId, authClientToken);
+        if (!JsonKey.UNAUTHORIZED.equals(clientId)) {
+          ctx.flash().put(JsonKey.AUTH_WITH_MASTER_KEY, Boolean.toString(true));
+        }
+      }
+      return clientId;
+    } else {
+      return JsonKey.ANONYMOUS;
+    }
+  }
+
+  /**
+   * this method will check incoming request required validation or not. if this method return true
```

Comment:
Checks if request URL is in excluded (i.e. public) URL list or not

requestUrl Request URL

True if URL is in excluded (public) URLs. Otherwise, returns false.